scripts/generating_yaml.py

```diff
@@ -0,0 +1,213 @@
+filter_ID_code = interactive(d, filter_ID_code=filter_ID);
+serial_into_USBhub = data['serial_into_USBhub_port']
+serial_into_USBhub_port = interactive(e, serial_into_USBhub_port=serial_into_USBhub);
+camera_into_serial = data['camera_into_serial_port']
+camera_into_serial_port = interactive(f, camera_into_serial_port=camera_into_serial);
+USBhub = data['USBhub_SN']
+USBhub_SN = interactive(j, USBhub_SN=USBhub);
+camera_into_USBhub = data['camera_into_USBhub_port']
+camera_into_USBhub_port = interactive(k, camera_into_USBhub_port=camera_into_USBhub);
+
+
+# In[9]:
+
+def h(add_device):
+    birger_SN = birger_serial_number.result
+    sbig_SN = sbig_serial_number.result
+    lens_SN = lens_serial_number.result
+    filter_ID = filter_ID_code.result
+    serial_to_USBhub_port = serial_into_USBhub_port.result
+    camera_to_serial_port = camera_into_serial_port.result
+    USB_hub_SN = USBhub_SN.result
+    camera_to_USBhub_port = camera_into_USBhub_port.result
+    additional_device = {'model' : 'sbig',
+                                                    'port' : sbig_SN,
+                                                   'filter_type' : filter_ID,
+                                                    'focuser' : {'model' : 'birger',
+                                                                  'port' : birger_SN
+
+                                                    },
+                                                    'lens' : {'model' : 'canon',
+                                                               'port' : lens_SN},
+                                                    'USB_hub_serial_number' : USB_hub_SN,
+                                                    'camera_into_serial_adaptor_port' : camera_to_serial_port,
+                                                    'serial_adaptor_into_USBhub_port' : serial_to_USBhub_port,
+                                                   'camera_into_USBhub_port' : camera_to_USBhub_port
+                                                              }
+    if data_list['cameras']['devices'] == [None]:
+        data_list['cameras']['devices'] = [additional_device]
+    else:
+        data_list['cameras']['devices'].append(additional_device)
+    return (add_device)
+
+
+# In[10]:
+
+date_info = datetime.datetime.today()
+datetime_str = date_info.strftime('%Y_%m_%d_%H_%M')
+
+archive_filename = '{}_{}.{}'.format('huntsman', datetime_str, 'yaml')
+
+
+# In[11]:
+
+archive_filename
+
+
+# In[12]:
+
+def g(create_file):
+    strOutFile1 = os.path.join("/var/huntsman-pocs/conf_files", 'huntsman.yaml')
+    objFile1 = open(strOutFile1, "w")
+    yaml.dump(data_list,objFile1, default_flow_style=False, indent=4)
+    objFile1.close()
+
+    strOutFile = os.path.join("/var/huntsman-pocs/conf_files/huntsman_archive", archive_filename)
+    objFile = open(strOutFile, "w")
+    yaml.dump(data_list,objFile, default_flow_style=False, indent=4)
+    objFile.close()
+
+    return create_file
+
```

Comment:
There is also quite a bit of white space, which you can fix by doing something I learned called pep8 linting. The easiest way I find is by running [autopep8](https://pypi.python.org/pypi/autopep8). You can just run 
```
pip install --upgrade autopep8
```
And then with a terminal running in your environment 
```
autopep8 --in-place --aggressive --aggressive <filename>
```